### ISSUE TITLE
Issue with "<animate>": Firefox is inconsistent in rendering svg - graphic

### ISSUE THREAD

**Issue by danilw** [mdn/content#8551]
MDN URL: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/animate

#### What information was incorrect, unhelpful, or incomplete?

https://developer.mozilla.org/en-US/docs/Web/SVG/Element/image
https://developer.mozilla.org/en-US/docs/Web/SVG/Element/animateMotion

And animate - everywhere Firefox has "green" full support, when on practice Firefox is inconsistent in rendering svg - graphic.

#### Specific section or headline?

Using animate (any of it) to apply transform on the image - image rendering for transform does not look like expected on Firefox. Smooth transform applied only to SVG shapes, except image.

#### What did you expect to see?

I expect to see a note or some warning that Firefox is inconsistent in rendering SVG graphic.

#### Did you test this? If so, how?

[motion_test.svg](https://danilw.github.io/GLSL-howto/motion_test_svg/motion_test.svg)

**gif from firefox** - image "jump" instead of smooth movements:

![ff](https://user-images.githubusercontent.com/24825887/131783368-cb165405-b002-4bde-9e08-35b1e36e5395.gif)

**expected** - no jump and smooth movement:

![c](https://user-images.githubusercontent.com/24825887/131783449-52d1fb1b-200f-4954-842d-f578bc22a5c9.gif)

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/svg/element/animate`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/SVG/Element/animate
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/svg/element/animate/index.html
* Last commit: https://github.com/mdn/content/commit/91456ead4017ad5b6c5be3cacc83cc72007483ca
* Document last modified: 2021-06-10T20:25:11.000Z

</details>

**Comment by vlwkaos** [comment]
I can confirm image svg element making a discrete movement from the example provided.

I did a further research on this topic and have read/skimmed: various documents.

> And animate - everywhere Firefox has "green" full support, when on practice Firefox is inconsistent in rendering svg - graphic.

While I am not completely familiar with these, I concluded that compatibility table status "green" for full support is not wrong per https://www.w3.org/TR/SVG11/conform.html#ConformingSVGViewers. The specification for conforming SVG Viewer does not include smooth artifact-free animation. 

However, I do agree that the animation related behavior should at least be consistent across 2d media(image, svg ...), and there should be some warning as you suggested.

I searched whether there were any open issue open issue about this in firefox development and found this: https://bugzilla.mozilla.org/show_bug.cgi?id=739176 While this issue is not about svg itselt, I inferred that this might be a problem with how firefox's image rendering. (Note that I am not completely sure about this). Maybe this issue could be linked to the pages you provided.

---

These are other documents / pages, I think, relevant to the topic.

- https://developer.mozilla.org/en-US/docs/Web/SVG/SVG_1.1_Support_in_Firefox
- https://www.w3.org/TR/SVG11/struct.html#ImageElement
- https://developer.mozilla.org/en-US/docs/Web/SVG/Element/animate
- https://www.w3.org/TR/2001/REC-smil-animation-20010904/
- https://github.dev/mozilla/gecko-dev/blob/d36cf98aa85f24ceefd07521b3d16b9edd2abcb7/dom/svg/SVGMotionSMILType.h

**Comment by sideshowbarker** [comment]
Howdy @jakearchibald — Does the problem in the issue description here seem like the same problem you reported at https://bugzilla.mozilla.org/show_bug.cgi?id=739176 ?

**Comment by jakearchibald** [comment]
Yep, that's the one. Gotta add a tiny rotate if you want it to animate smoothly in Firefox.

It isn't just an SVG issue, it's a general animation issue.

**Comment by danilw** [comment]
> Yep, that's the one. Gotta add a tiny rotate if you want it to animate smoothly in Firefox.
> 
> It isn't just an SVG issue, it's a general animation issue.

Even more reasons to make it as Note in SVG animate page about Firefox and Image, especially if this "bug" is 10yo and not going to be fixed.

**Comment by sideshowbarker** [comment]
Thanks @jakearchibald for confirming, and thanks @vlwkaos for doing the investigation.

> Even more reasons to make it as Note in SVG animate MDN page about Firefox and Image, especially if this "bug" is 10yo and not going to be fixed.

Yeah it’s clear that we need to do more here to give developers a heads-up. So I’ve assigned myself for now.

@vlwkaos Would you be interested in writing up a PR for this? If not, I can try to put together a PR with a change for this myself.

**Comment by vlwkaos** [comment]
@sideshowbarker I will try and come up with a change. Also I am thinking of testing it one more time with non-svg image animation to confirm the linked issue.

**Comment by sideshowbarker** [comment]
> @sideshowbarker Okay, I will try and come up with a change.

Super — then I’m re-assigning this to you.

So I think minimally the change/addition should include some information about the rotate hack that Jake mentions, and that’s mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=739176#c25, etc.

And not to complicate things for you too much — but I think ideally the change/addition should maybe also include some info about the rotate hack possibly not being effective in some cases, as mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=739176#c27, but that in those cases there’s still at least one other way to hack around it, as mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=739176#c29 and https://bugzilla.mozilla.org/show_bug.cgi?id=739176#c30

**Comment by jakearchibald** [comment]
> about the rotate hack possibly not being effective in some cases, as mentioned in https://bugzilla.mozilla.org/show_bug.cgi?id=739176#c27

Fwiw, that's a layout-driven animation, so it's kinda different to the others.

Unfortunately the Firefox issue has gotten worse as it'll sub-pixel translate parts of an element, but not all of it. https://www.youtube.com/watch?v=C665W2nwECk (the bottom version has the rotate hack).

**Comment by vlwkaos** [comment]
@sideshowbarker I am trying to decide where to put the relevant information since this turned out to be more than just SVG animation issue. Any help would be much appreciated.

**Comment by sideshowbarker** [comment]
> @sideshowbarker I am trying to decide where to put the relevant information since this turned out to be more than just SVG animation issue. Any help would be much appreciated.

I don’t have any particular insight to be able to give useful guidance on that.

**Comment by hamishwillee** [comment]
@teoli2003 You got any thoughts on ^^^

**Comment by Josh-Cena** [comment]
Actually I wasn't able to notice a difference on the latest Firefox for any of the demos (Firefox 136, macOS). Perhaps it's a me problem. But in general we don't document browser bugs, and this surely is very hard to precisely describe, so I think we should leave it to developers to figure out and perhaps better resources exist elsewhere.

**Comment by danilw** [comment]
> Actually I wasn't able to notice a difference on the latest Firefox for any of the demos (Firefox 136, macOS). Perhaps it's a me problem.

Linux - **same as before** firefox 135.0. 
_(still pixel jumps)_

Video (made just now):

https://github.com/user-attachments/assets/0e0b2056-772b-42d3-8d83-aef6383c8416

Top - Firefox 135 (linux)
Bot - Chrome
It clearly "visible"

**Link to example** [**svg**](https://danilw.github.io/GLSL-howto/motion_test_svg/motion_test.svg) from first my message.

Other example where I originally found this behavior -  https://danilw.github.io/GLSL-howto/y.svg
(bottom image movements slow and smooth in Chrome - and pixel jumps in Chome)
